commit/chainfee/observation.go

```diff
@@ -28,10 +30,36 @@ func (p *processor) Observation(
 ) (Observation, error) {
 	lggr := logutil.WithContextValues(ctx, p.lggr)
 
-	feeComponents := p.obs.getChainsFeeComponents(ctx, lggr)
-	nativeTokenPrices := p.obs.getNativeTokenPrices(ctx, lggr)
-	chainFeeUpdates := p.obs.getChainFeePriceUpdates(ctx, lggr)
-	fChain := p.observeFChain(lggr)
+	var (
+		feeComponents     map[cciptypes.ChainSelector]types.ChainFeeComponents
+		nativeTokenPrices map[cciptypes.ChainSelector]cciptypes.BigInt
+		chainFeeUpdates   map[cciptypes.ChainSelector]Update
+		fChain            map[cciptypes.ChainSelector]int
+	)
+
+	operations := []func(ctx context.Context, l logger.Logger){
+		func(ctx context.Context, l logger.Logger) { feeComponents = p.obs.getChainsFeeComponents(ctx, l) },
+		func(ctx context.Context, l logger.Logger) { nativeTokenPrices = p.obs.getNativeTokenPrices(ctx, l) },
+		func(ctx context.Context, l logger.Logger) { chainFeeUpdates = p.obs.getChainFeePriceUpdates(ctx, l) },
+		func(ctx context.Context, l logger.Logger) { fChain = p.observeFChain(l) },
+	}
+
+	callTimeout := p.cfg.ChainFeeAsyncObserverSyncTimeout
+	callCtx, cf := context.WithTimeout(ctx, callTimeout)
+	defer cf()
+	lggr.Debugw("spawning observing goroutines", "timeout", callTimeout)
+
+	wg := sync.WaitGroup{}
+	wg.Add(len(operations))
+	for i, op := range operations {
```

Comment:
How would that behave in case of a timeout? Are we going to continue processing with partially initialized values (feeComponents, nativeTokenPrices etc).

Can we have a test for that logic with timeouts hitting / not hitting different processors?

Reply:
Thats a good question, I think in the case of timeout for all these funcs that have ctx we will log the error (ctx deadline exceeded?) and just return an empty map / array

Reply:
> This concern though is not related to the changes of this PR, I'd rather not overcomplicat it with test scenarios not related to it.

Are you sure concern is not valid? Before we had a timeout applied on the OCR level, in case of a timeout, we break execution immediately and cancel the entire observation. Right now, we are terminating on a lower level and code keeps continuing to work, right? Therefore, after the timeout, we will have data that is partially initialized and might lead to some weird behaviour. This is new behaviour. I think tests are needed here